scripts/popup.js

```diff
@@ -11,18 +11,21 @@
         ls.folder_name = $.trim(this.value);
       });
 
+    // Register file renaming listener
+    $('#file_renaming_textbox')
+      .val(ls.new_file_name)
+      .on('change', function () {
+        ls.new_file_name = $.trim(this.value);
+      });
+
     // Register filter URL listener
     $('#filter_textbox')
       .val(ls.filter_url)
       .on('change', function () {
         ls.filter_url = $.trim(this.value);
       });
 
-    chrome.downloads.onDeterminingFilename.addListener(function (item, suggest) {
-      if (ls.folder_name) {
-        suggest({ filename: ls.folder_name + '/' + item.filename});
-      }
-    });
+    chrome.downloads.onDeterminingFilename.addListener(suggestNewFilename);
```

Comment:
Extracted this to a separate function. My functions are way too long right now :) One of my goals is to start adding tests soon, then refactor bit by bit.